Contact form - cu-jjvezd

cameronbaney opened the pull request:
# Description
The purpose of this PR is to add contact forms to the portal. These two forms are for general inquiries and for reporting a bug. For general inquiries, it will send an email to SPARC. For bugs, it will create a wrike ticket.

## Tickets
https://app.clickup.com/t/jjvezd
https://www.wrike.com/open.htm?id=467362300

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- In the header, click on "Contact Us'
- You should be taken to the contact us page
- Select `I couldn’t find the information in Help` from the dropdown
- The general form should be shown
- Clicking submit should notify you what fields are required
- Fill out the form and click "submit"
  - Checking "Subscribe to the SPARC Newsletter" should add that email to the Mailchimp list. Ask Leo for access to verify.
- You should now see a success message with your first name displayed
- Click on "Submit another inquiry"
- Select `I want to report an error or a technical issue` from the dropdown
- Clicking submit should notify you what fields are required
- Fill out the form and click "submit"
- You should now see a success message with your first name displayed
- Go to Wrike and verify that you ticket has been created, and the ticket's body has the correct info you added in the form. Protip: Search for your ticket's name, which is the value you entered for "Provide a short description of what you were doing.*"

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests that prove my fix is effective or that my feature works

mahamzayed commented:
I get a `500 Internal Server Error` when trying to submit the contact form. This is the endpoint that's called `http://localhost:8000/contact`. I made sure `sparc-api` was running before I tested.

mahamzayed reviewed (commented):
Code LGTM. Just waiting to verify the functionality flow once endpoint requests work.

cameronbaney commented:
@mahamzayed You do not need `sparc-api` to be running locally as the version on dev is the latest to support these features. Please run through this branch while using the dev `sparc-api` while this issue is being looked on.

I can verify I also get these errors while running the `sparc-api` locally, and  am looping @JeremyLecoeur in about that. The `500` error received is: 
```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>
```

mahamzayed commented:
@cameronbaney Also I forgot to note - when you view the web app in a responsive/mobile view, the link to the contact us page takes you to the old wrike form.

cameronbaney commented:
@mahamzayed thank you very much for that. i noticed that as well when testing out the issue you found with the sparc-api and fixed it with 4ae5b7c

mahamzayed commented:
![image](https://user-images.githubusercontent.com/6557946/109984396-b87ad980-7cd1-11eb-82fe-2a82e3156ee6.png)

There's a stray comma in the success message after submitting the `I couldn't find information in Help` form.

mahamzayed commented on components/ContactUsForms/GeneralForm/GeneralForm.vue:
```diff
@@ -0,0 +1,217 @@
+      />
+    </el-form-item>
+
+    <!-- <el-form-item prop="shouldSendCopy">
+      <el-checkbox v-model="form.shouldSendCopy">
+        Please send me a copy of this message
+      </el-checkbox>
+    </el-form-item> -->
+
```
Remove

mahamzayed reviewed (approved):
Functionality works as expected except the minor things I mentioned. I couldn't test the mailchimp feature because the endpoint was giving me a `404` which I'm assuming from @JeremyLecoeur's suggestion is because my email address is already added to the mailing list. I'll confirm with Leo to check.

mahamzayed commented:
@cameronbaney @JeremyLecoeur It looks like the mailchimp endpoint is erroring out when I try to subscribe to the newsletter:

```
curl 'https://sparc-api.herokuapp.com/mailchimp' \
  -H 'Connection: keep-alive' \
  -H 'Pragma: no-cache' \
  -H 'Cache-Control: no-cache' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.150 Safari/537.36' \
  -H 'Content-Type: application/json;charset=UTF-8' \
  -H 'Origin: http://localhost:3000' \
  -H 'Sec-Fetch-Site: cross-site' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Referer: http://localhost:3000/' \
  -H 'Accept-Language: en-US,en;q=0.9,la;q=0.8' \
  --data-raw '{"email_address":"maha+1234@blackfynn.com"}' \
  --compressed
  ```
  It's giving me a `404 Bad Request`

cameronbaney commented:
@mahamzayed Thanks for reviewing. I resolved most of the issues you brought up, except for clicking "Contact Us" to reset the form. This is because the user isn't navigating away from the Contact Us page, so it's not triggering any route changes. I don't think this functionality is expected, but I can think of some solutions if you deem this a necessary feature.

Thanks for finding that newsletter issue. Can you let me know which form you experienced that on?

mahamzayed commented:
> @mahamzayed Thanks for reviewing. I resolved most of the issues you brought up, except for clicking "Contact Us" to reset the form. This is because the user isn't navigating away from the Contact Us page, so it's not triggering any route changes. I don't think this functionality is expected, but I can think of some solutions if you deem this a necessary feature.
> 
> Thanks for finding that newsletter issue. Can you let me know which form you experienced that on?
> 
> edit: Nevermind, it's happening on both forms.

Its not a blocker for deployment but I think it would improve user functionality if a solution was implemented for it. If it's not going to take up too much time, I don't think it hurts to add a trigger for a route change.

cameronbaney commented:
@mahamzayed Thank you for the feedback. I have addressed the issues you found. For the newsletter issue, the first name and last name fields will now be required only if the user is subscribing to the newsletter. 

In addition, I added the feature to reset the form when a user navigates to Contact Us via the link. This also allows users to now link directly to a form if they want to.

mahamzayed reviewed (approved):
Looks great and works as expected!
